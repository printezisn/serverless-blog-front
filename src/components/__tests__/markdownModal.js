import React from "react"
import renderer from "react-test-renderer"

import MarkdownModal from "../markdownModal"

jest.mock("../markdown", () => "Markdown")

describe("MarkdownModal", () => {
    it("renders correctly when it's not active", () => {
        const modal = renderer
            .create(
                <MarkdownModal
                    isActive="false"
                    text="Test inactive text."
                    onClose={() => {}}
                />
            )
            .toJSON()
        expect(modal).toMatchSnapshot()
    })

    it("renders correctly when it's active", () => {
        const modal = renderer
            .create(
                <MarkdownModal
                    isActive="true"
                    text="Test active text."
                    onClose={() => {}}
                />
            )
            .toJSON()
        expect(modal).toMatchSnapshot()
    })

    it("closes when the close button is clicked", () => {
        let totalInvokes = 0

        const modal = renderer.create(
            <MarkdownModal
                isActive="true"
                text="Test active text."
                onClose={() => totalInvokes++}
            />
        )

        modal.root.findByProps({ className: "delete" }).props.onClick()
        modal.root
            .findByProps({ className: "modal-card-foot" })
            .findByProps({ className: "button" })
            .props.onClick()

        expect(totalInvokes).toBe(2)
    })
})
