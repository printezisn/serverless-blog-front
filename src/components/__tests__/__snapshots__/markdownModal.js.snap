// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownModal renders correctly when it's active 1`] = `
<div
  className="modal is-active"
>
  <div
    className="modal-background"
  />
  <div
    className="modal-card"
  >
    <div
      className="modal-card-head"
    >
      <p
        className="modal-card-title"
      >
        Preview
      </p>
      <button
        aria-label="close"
        className="delete"
        onClick={[Function]}
      />
    </div>
    <div
      className="modal-card-body"
    >
      <Markdown
        text="Test active text."
      />
    </div>
    <div
      className="modal-card-foot"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <i
          className="fas fa-arrow-left"
        />
          Close
      </button>
    </div>
  </div>
</div>
`;

exports[`MarkdownModal renders correctly when it's not active 1`] = `
<div
  className="modal"
>
  <div
    className="modal-background"
  />
  <div
    className="modal-card"
  >
    <div
      className="modal-card-head"
    >
      <p
        className="modal-card-title"
      >
        Preview
      </p>
      <button
        aria-label="close"
        className="delete"
        onClick={[Function]}
      />
    </div>
    <div
      className="modal-card-body"
    >
      <Markdown
        text="Test inactive text."
      />
    </div>
    <div
      className="modal-card-foot"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <i
          className="fas fa-arrow-left"
        />
          Close
      </button>
    </div>
  </div>
</div>
`;
